lib/elixir/test/elixir/kernel/warning_test.exs

```diff
@@ -1768,6 +1768,65 @@ defmodule Kernel.WarningTest do
     purge(TestMod)
   end
 
+  test "deprecate remote nullary zero-arity calls without parens" do
+    assert capture_err(fn ->
+             Code.eval_string("""
+             defmodule TestMod do
+                def foo(), do: nil
+             end
```

Comment:
There is no need to define a new module. Instead, let's use a zero arity function in Elixir's codebase, such as `System.pid`.